sw/airborne/modules/computer_vision/detect_gate.c

```diff
@@ -109,11 +139,52 @@ struct image_t *detect_gate_func(struct image_t *img)
   } else {
     // perform snake gate detection:
     snake_gate_detection(img, n_samples, min_px_size, min_gate_quality, gate_thickness, min_n_sides, color_Ym, color_YM,
-                         color_Um, color_UM, color_Vm, color_VM);
+                         color_Um, color_UM, color_Vm, color_VM, &best_gate);
+
+    /*
+    // debugging snake gate:
+    printf("Detected gate: ");
+    for(int i = 0; i < 4; i++) {
+      printf("(%d,%d) ", best_gate.x_corners[i], best_gate.y_corners[i]);
+    }
+    printf("\n");
+    */
+
+    drone_position = get_world_position_from_image_points(best_gate.x_corners, best_gate.y_corners, world_corners, 3,
+                     DETECT_GATE_CAMERA.camera_intrinsics, cam_body);
+    drone_position.x -= gate_dist_x;
+
+    // debugging the drone position:
+    // printf("Position drone: (%f, %f, %f)\n", drone_position.x, drone_position.y, drone_position.z);
+
+    // send from thread to module
+    pthread_mutex_lock(&gate_detect_mutex);
+    detect_gate_x = drone_position.x;
+    detect_gate_y = drone_position.y;
+    detect_gate_z = drone_position.z;
+    detect_gate_has_new_data = true;
+    pthread_mutex_unlock(&gate_detect_mutex);
   }
   return img;
 }
 
+void detect_gate_event(void)
+{
+  static int32_t cnt = 0;
+  pthread_mutex_lock(&gate_detect_mutex);
+  if (detect_gate_has_new_data) {
+    detect_gate_has_new_data = false;
+    AbiSendMsgRELATIVE_LOCALIZATION(DETECT_GATE_ABI_ID, cnt++,
```

Comment:
When sending an ABI message, it shouldn't be with the broadcast value, but with a unique ID (for this type of message) from `subsystems/abi_sender_ids.h`.
The `ABI_BROADCAST`should only be used when binding to a message without filtering on the sender.

Reply:
I've added an ABI id to the abi_sender_ids.h file